n2y/plugins/internallinks.py

```diff
@@ -0,0 +1,68 @@
+import typing
+import uuid
+import urllib.parse
+
+from n2y.blocks import Block, ChildPageBlock
+from n2y.logger import logger
+from n2y.page import Page
+from n2y.rich_text import TextRichText
+from n2y.utils import header_id_from_text
+
+
+def get_notion_id_from_href(href: str) -> typing.Optional[str]:
+    """Extract the ID of a target block from a href fragment"""
+    target_id = urllib.parse.urlparse(href, allow_fragments=True).fragment
+    if target_id is None:
+        return None
+    # Convert to UUID and back to ensure the format is correct
+    try:
+        return str(uuid.UUID(target_id))
+    except ValueError:
+        # fragment is not a Notion ID
+        return None
+
+
+def is_internal_link(href: typing.Optional[str], notion_id: str) -> bool:
+    """Is the href fragment a link to a block on the same page?"""
+    return href is not None and href.startswith(f"/{notion_id.replace('-', '')}")
+
+
+def find_target_block(page: Page, target_id: str) -> Block:
+    page_block: ChildPageBlock = page.block
+    if page_block is None:
+        raise ValueError("Page.block is None")
+
+    try:
+        return next(
```

Comment:
What if the `target_id` is to a child of one of the `ChildPageBlock's` children? Or the child of one of its children's children? This might need to be recursive logic.

Reply:
Good catch. Can you give some concrete examples where this would occur so I can understand the usecase more (I don't want to over generalize too soon)? Like a heading inside of a ___ ?

Reply:
So, the `ListItemBlock` objects would be a good example as they are really just a serialization container for rich text arrays. They may also have `TableBlock`, `ParagraphBlock`, or `ListItemBlock` children of their own. Speaking of, `ParagraphBlock` objects can be children of each other. `TableBlock` objects have `RowBlock` children. `TableOfContentsBlock` objects have `TableOfContentsItemBlock`. `ToggleBlock` objects work similar to `ListItemBlock` objects, I believe. Lastly, I think `ColumnListBlock` object's children are `ColumnBlock` objects, which will be able to have children of all the same type as the `ListItemBlock` objects.

Reply:
Updated here: https://github.com/innolitics/n2y/pull/195/commits/08f29d0a604cda9aa95defb310e08e8d2f4b3da4